torch/distributed/_tensor/debug/visualize_sharding.py

```diff
@@ -0,0 +1,169 @@
+from typing import List, Sequence, Tuple
+from torch._prims_common import ShapeType
+from torch.distributed._tensor import DeviceMesh, Shard
+from torch.distributed._tensor.placement_types import (
+    Placement,
+    Shard,
+)
+import numpy as np
+
+def _mesh_to_coordinate(mesh, device_type):
+    '''
+    Given a n-dimensional list of device mesh, this function creates a map of
+    device and its coordinate
+    '''
+    # Convert the n-dimensional list to a NumPy array
+    np_mesh = np.array(mesh.mesh.tolist())
+
+    # Create a dictionary to map each value to its coordinate
+    device_to_coordinate_map = {}
+    for coord, value in np.ndenumerate(np_mesh):
+        # device is unique in device_mesh
+        device_to_coordinate_map[f"{device_type}:{str(value)}"] = list(coord)
+
+    return device_to_coordinate_map
+
+def _convert_offset_to_ranges(all_offsets):
+    '''
+    Using tabulate package to create a table is easier when we specify row and col ranges
+    This function converts offsets to ranges.
+    '''
+    converted_blocks = []
+
+    for offset in all_offsets:
+        shape, offset, value = offset
+
+        # Calculate row_range and column_range
+        row_range = (offset[0], offset[0] + shape[0] - 1)
+        column_range = (offset[1], offset[1] + shape[1] - 1)
+
+        # Convert value to string to match your desired format
+        converted_block = {
+            'row_range': row_range, 
+            'column_range': column_range, 
+            'value': str(value)
+        }
+        converted_blocks.append(converted_block)
+
+    return converted_blocks
+
+def _create_table(blocks):
+    '''
+    Creates a tabulate table given row and column ranges with device name
+    '''
+    try:
+        from tabulate import tabulate
+    except ImportError:
+        raise ImportError("tabulate package is required to visualize sharding")
+
+    # Extract unique row and column ranges
+    row_ranges = sorted(set([block['row_range'] for block in blocks]))
+    col_ranges = sorted(set([block['column_range'] for block in blocks]))
+
+    # Create a matrix initialized with empty strings
+    matrix = [['' for _ in col_ranges] for _ in row_ranges]
+
+    # Fill the matrix with values
+    for block in blocks:
+        row_index = row_ranges.index(block['row_range'])
+        col_index = col_ranges.index(block['column_range'])
+        if matrix[row_index][col_index] == '':
+            matrix[row_index][col_index] = block['value']
+        else:
+            matrix[row_index][col_index] += ', ' + block['value']
+
+    # Prepare headers
+    row_headers = [f"Row {r[0]}-{r[1]}" for r in row_ranges]
+    col_headers = [f"Col {c[0]}-{c[1]}" for c in col_ranges]
+
+    return tabulate(matrix, headers=col_headers, showindex=row_headers)
+
+def compute_local_shape_and_global_offset(
```

Comment:
hmmm wait, this is exactly the same implementation as https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/_utils.py#L48? If so, I think we should reuse the util instead of duplicating

Reply:
No, I changed the implementation slightly to take `my_coordinate` as parameter and remove this line: https://github.com/pytorch/pytorch/blob/0fc04e274de9d6a52172674e294c1b73d43a50e9/torch/distributed/_tensor/_utils.py#L88

I will update the docs as mentioned above

Reply:
hmmm i see, but I didn't see where you are using the my_coordinate that is different from `mesh.get_coordinate()` inside this util? could you explain why you need to pass in my_coordinate again?

Reply:
By default the function `compute_local_shape_and_global_offset` returns the coordinate for only the current device. We need to get all coordinates on device 0 for the visualization. So I use the device mesh and for each element in the mesh, I call this function with different `my_coordinate` to get the overall tensor splits. Is that the right approach or can we optimize it further?

Reply:
Oh I see that's the device_map logic you did below, make sense now.